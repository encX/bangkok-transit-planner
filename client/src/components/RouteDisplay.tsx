import React, {FunctionComponent} from 'react';

interface RouteDisplayProps {
	test?: string;
}

export const RouteDisplay: FunctionComponent<RouteDisplayProps> = (props: RouteDisplayProps): JSX.Element => {
	return (
		<div/>
	);
};
