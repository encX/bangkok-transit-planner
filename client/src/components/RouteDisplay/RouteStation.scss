@import "src/css/Common";

.route-station {
	&--time {
		padding-top: $station-dot-pad-top;
		text-align: right;
	}
}
