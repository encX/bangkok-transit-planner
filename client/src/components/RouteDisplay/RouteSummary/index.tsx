import React, {FC} from 'react';
import {RouteSummaryItem} from './RouteSummaryItem';

interface RouteSummaryProps {
	cost: number;
	time: string;
	transfer: number;
}

const RouteSummary: FC<RouteSummaryProps> = ({cost, time, transfer}): JSX.Element => {
	return (
		<>
			<RouteSummaryItem name="Fare" value={`${cost} Baht`}/>
			<RouteSummaryItem name="Time" value={time}/>
			<RouteSummaryItem name="Transfer" value={`${transfer} times`}/>
		</>
	);
};

export default RouteSummary;
