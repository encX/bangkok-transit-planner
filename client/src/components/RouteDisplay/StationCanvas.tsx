import React, {FC} from 'react';

import './StationCanvas.scss';

interface StationCanvasProps {
	mode: 'start' | 'middle' | 'end';
	color?: string;
}

export const StationCanvas: FC<StationCanvasProps> = ({mode, color}): JSX.Element => {
	return (
		<div className="station-canvas">
			<div className={`station-canvas--top station-canvas--top-${mode}`} style={{backgroundColor: color}}/>
			<div className={`station-canvas--bottom station-canvas--bottom-${mode}`} style={{backgroundColor: color}}/>
			<div className="station-canvas--dot" style={{borderColor: color}}/>
		</div>
	);
};
