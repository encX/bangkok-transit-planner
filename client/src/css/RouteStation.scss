@import "Common";

.route-station {
	&--time {
		padding-top: $station-dot-pad-top;
		text-align: right;
	}
}
