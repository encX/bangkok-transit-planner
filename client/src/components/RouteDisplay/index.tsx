import React, {FunctionComponent} from 'react';

import {Route} from '../../api/models/Route';
import RouteSummary from './RouteSummary';
import {RouteStation} from './RouteStation';

const RouteDisplay: FunctionComponent<Route> = ({stations, cost}: Route): JSX.Element => {
	const start = stations.slice(0, 1)[0];
	const between = stations.slice(1, -1);
	const end = stations.slice(-1)[0];
	
	return(
		<div className="route-display">
			<RouteSummary cost={cost} time="0 min" transfer={0}/>
			<RouteStation mode="start" stationName={start.name}/>
			{between.map((station, i) => <RouteStation mode="middle" key={i} stationName={station.name}/>)}
			<RouteStation mode="end" stationName={end.name}/>
		</div>
	);
};

export default RouteDisplay;
