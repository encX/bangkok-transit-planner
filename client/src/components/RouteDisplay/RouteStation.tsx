import {Col, Row, Typography} from 'antd';
import React, {FunctionComponent} from 'react';
import {StationCanvas} from './StationCanvas';

import './RouteStation.scss';

const {Title} = Typography;

interface RouteStationProps {
	mode: 'start' | 'middle' | 'end';
	stationName: string;
}

export const RouteStation: FunctionComponent<RouteStationProps> = ({mode, stationName}): JSX.Element => {
	return (
		<Row gutter={[16, 0]}>
			<Col flex="90px"><div className="route-station--time">11:11 AM</div></Col>
			<Col flex="40px"><StationCanvas mode={mode} /></Col>
			<Col flex="auto"><Title>{stationName}</Title></Col>
		</Row>
	);
};
