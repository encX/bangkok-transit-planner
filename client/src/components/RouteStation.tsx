import {Col, Row} from 'antd';
import React, {FunctionComponent} from 'react';
import {StationCanvas} from './StationCanvas';

interface RouteStationProps {
	mode: 'start' | 'middle' | 'end';
	stationName: string;
}

export const RouteStation: FunctionComponent<RouteStationProps> = ({mode, stationName}): JSX.Element => {
	return (
		<Row gutter={[16, 0]}>
			<Col flex="80px">11:11am</Col>
			<Col flex="40px"><StationCanvas mode={mode} /></Col>
			<Col flex="auto"><h1>{stationName}</h1></Col>
		</Row>
	);
};
