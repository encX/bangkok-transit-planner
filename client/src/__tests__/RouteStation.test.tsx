import React from 'react';
import {shallow} from 'enzyme';
import {RouteStation} from '../components/RouteDisplay/RouteStation';

describe('RouteStation', function() {
	it('case "start": renders correctly', () => {
		const tree = shallow(<RouteStation mode="start" stationName="station"/>);
		expect(tree).toMatchSnapshot();
	});
	
	it('case "middle": renders correctly', () => {
		const tree = shallow(<RouteStation mode="middle" stationName="station"/>);
		expect(tree).toMatchSnapshot();
	});
	
	it('case "end": renders correctly', () => {
		const tree = shallow(<RouteStation mode="end" stationName="station"/>);
		expect(tree).toMatchSnapshot();
	});
});
