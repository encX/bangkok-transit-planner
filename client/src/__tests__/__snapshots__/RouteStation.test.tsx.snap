// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteStation case "end": renders correctly 1`] = `
<Row
  gutter={
    Array [
      16,
      0,
    ]
  }
>
  <Col
    flex="90px"
  >
    <div
      className="route-station--time"
    >
      11:11 AM
    </div>
  </Col>
  <Col
    flex="40px"
  >
    <StationCanvas
      mode="end"
    />
  </Col>
  <Col
    flex="auto"
  >
    <Title>
      station
    </Title>
  </Col>
</Row>
`;

exports[`RouteStation case "middle": renders correctly 1`] = `
<Row
  gutter={
    Array [
      16,
      0,
    ]
  }
>
  <Col
    flex="90px"
  >
    <div
      className="route-station--time"
    >
      11:11 AM
    </div>
  </Col>
  <Col
    flex="40px"
  >
    <StationCanvas
      mode="middle"
    />
  </Col>
  <Col
    flex="auto"
  >
    <Title>
      station
    </Title>
  </Col>
</Row>
`;

exports[`RouteStation case "start": renders correctly 1`] = `
<Row
  gutter={
    Array [
      16,
      0,
    ]
  }
>
  <Col
    flex="90px"
  >
    <div
      className="route-station--time"
    >
      11:11 AM
    </div>
  </Col>
  <Col
    flex="40px"
  >
    <StationCanvas
      mode="start"
    />
  </Col>
  <Col
    flex="auto"
  >
    <Title>
      station
    </Title>
  </Col>
</Row>
`;
